pkg/controller/monitor/monitor_controller.go

```diff
@@ -225,6 +231,30 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 			return reconcile.Result{}, err
 		}
 	}
+	preDefaultPatchFrom := client.MergeFrom(instance.DeepCopy())
+	fillDefaults(instance)
+	// Patch the monitor resource with defaults added.
+	if err = r.client.Patch(ctx, instance, preDefaultPatchFrom); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Failed to write defaults", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	var externalPrometheus bool
+	if instance.Spec.ExternalPrometheus != nil {
+		if err = r.client.Get(ctx, client.ObjectKey{Name: instance.Spec.ExternalPrometheus.Namespace}, &corev1.Namespace{}); err != nil {
+			if !errors.IsNotFound(err) {
+				r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get external prometheus namespace", err, reqLogger)
+				return reconcile.Result{}, err
+			}
+			// We set ExternalPrometheus to nil, and proceed to render the other configuration.
+			// When the namespace is created, we will reconcile the ExternalPrometheus.
+			reqLogger.V(3).
+				WithValues("namespace", instance.Spec.ExternalPrometheus.Namespace).
+				Info("will skip external prometheus configuration, namespace does not exist")
+			instance.Spec.ExternalPrometheus = nil
```

Comment:
I think I commented on this in the previous iteration of this PR, but I'm not comfortable overriding user input like this.

We should represent this state differently without trying to piggy-back on the API resource by modifying it (if we even want to do this at all). The more I think about it, the more I wonder why we don't just treat this as invalid configuration - the user is providing us input that isn't actionable, and that is generally considered an error.